device/samples/simple_sample_device.js

```diff
@@ -16,50 +16,61 @@ var Message = require('azure-iot-device').Message;
 //  "HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>"
 var deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
 
-// fromConnectionString must specify a transport constructor, coming from any transport package.
-var client = Client.fromConnectionString(deviceConnectionString, Protocol);
-
-var connectCallback = function (err) {
-  if (err) {
-    console.error('Could not connect: ' + err.message);
-  } else {
-    console.log('Client connected');
-    client.on('message', function (msg) {
-      console.log('Id: ' + msg.messageId + ' Body: ' + msg.data);
-      // When using MQTT the following line is a no-op.
-      client.complete(msg, printResultFor('completed'));
-      // The AMQP and HTTP transports also have the notion of completing, rejecting or abandoning the message.
-      // When completing a message, the service that sent the C2D message is notified that the message has been processed.
-      // When rejecting a message, the service that sent the C2D message is notified that the message won't be processed by the device. the method to use is client.reject(msg, callback).
-      // When abandoning the message, IoT Hub will immediately try to resend it. The method to use is client.abandon(msg, callback).
-      // MQTT is simpler: it accepts the message by default, and doesn't support rejecting or abandoning a message.
-    });
+async function disconnectHandler () {
+  clearInterval(sendInterval);
```

Comment:
Weren't we told not to send async to functions that weren't expecting them?  You don't send async's as any of the other callbacks.